frontend/eda/access/credentials/EditCredential.tsx

```diff
@@ -75,21 +82,48 @@ function CredentialInputs() {
         labelHelp={credentialTypeHelpBlock}
         labelHelpTitle={t('Credential type')}
       />
-      <PageFormTextInput<EdaCredentialCreate>
-        name="username"
-        label={t('Username')}
-        isRequired
-        placeholder={t('Enter username')}
-      />
-      <PageFormTextInput<EdaCredentialCreate>
-        name="secret"
-        label={t('Token/Password')}
-        type="password"
-        placeholder={t('Enter credential token or password')}
-        isRequired
-        labelHelp={t('Tokens/passwords allow you to authenticate to your destination.')}
-        labelHelpTitle={t('Token/Password')}
-      />
+      <PageFormHidden
+        watch="credential_type"
+        hidden={(type: CredentialTypeEnum) =>
+          type !== CredentialTypeEnum.GitHubPersonalAccessToken &&
+          type !== CredentialTypeEnum.GitLabPersonalAccessToken &&
+          type !== CredentialTypeEnum.ContainerRegistry
+        }
+      >
+        <PageFormTextInput<EdaCredentialCreate>
+          name="username"
+          label={t('Username')}
+          placeholder={t('Enter username')}
+        />
+        <PageFormTextInput<EdaCredentialCreate>
+          name="secret"
+          label={t('Token/Password')}
+          type="password"
+          placeholder={t('Enter credential token or password')}
+          isRequired
+          labelHelp={t('Tokens/passwords allow you to authenticate to your destination.')}
+          labelHelpTitle={t('Token/Password')}
+        />
+      </PageFormHidden>
+      <PageFormHidden
+        watch="credential_type"
+        hidden={(type: CredentialTypeEnum) => type !== CredentialTypeEnum.AnsibleVaultPassword}
+      >
+        <PageFormTextInput<EdaCredentialCreate>
+          name="key"
+          label={t('Vault identifier')}
+          placeholder={t('Vault identifier')}
+        />
+        <PageFormTextInput<EdaCredentialCreate>
+          name="secret"
+          label={t('Vault password')}
+          type="password"
+          placeholder={t('Enter vault password')}
+          isRequired
+          labelHelp={t('Vault password')}
+          labelHelpTitle={t('Vault password')}
+        />
```

Comment:
Can we get data-cy locators added to the elements in this file that we may need to be able to access with Cypress?